homework/index.js

```diff
@@ -29,23 +31,115 @@
     return elem;
   }
 
-  function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+  function renderRepoTable(table, name, path) {
+    const tr = createAndAppend('tr', table);
+    createAndAppend('th', tr, {
+      text: name,
+    });
+    createAndAppend('td', tr, {
+      text: path,
+    });
+    return tr;
+  }
+
+  function renderRepoDetails(repo, table) {
+    const firstRow = renderRepoTable(table, 'Repository', '');
+    createAndAppend('a', firstRow.lastChild, {
+      text: repo.name,
+      href: repo.html_url,
+      target: '_blank',
+    });
+    renderRepoTable(table, 'Description:', repo.description);
+    renderRepoTable(table, 'Forks:', repo.forks);
+    const date = new Date(repo.updated_at).toLocaleString('en-US');
+    renderRepoTable(table, 'Updated:', date);
+  }
+
+  function renderContributors(url, div) {
+    fetchJSON(url)
+      .then(repos => {
+        repos.forEach(repo => {
+          const divContributor = createAndAppend('div', div, {
+            class: 'contributor-details',
+          });
+          const linkContributor = createAndAppend('a', divContributor, {
+            href: repo.html_url,
+            target: '_blank',
+          });
+          createAndAppend('img', linkContributor, {
+            src: repo.avatar_url,
+            class: 'avatar',
+          });
+          createAndAppend('h4', linkContributor, {
+            text: repo.login,
+          });
+          createAndAppend('p', linkContributor, {
+            text: repo.contributions,
+            class: 'contribution-count',
+          });
+        });
+      })
+      .catch(err => {
+        createAndAppend('div', div, {
+          text: err.message,
+          class: 'alert-error',
+        });
+      });
   }
 
   function main(url) {
-    fetchJSON(url, (err, repos) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, {
+    fetchJSON(url)
+      .then(repos => {
+        const root = document.getElementById('root');
+        const header = createAndAppend('header', root, {
+          class: 'header-class',
+        });
+        createAndAppend('h3', header, {
+          text: 'HYF Repository',
+          class: 'title',
+        });
+        const select = createAndAppend('select', header, {
+          class: 'select-class',
+        });
+        const mainContainer = createAndAppend('section', root, {
+          class: 'main-container',
+        });
+        const reposContainer = createAndAppend('section', mainContainer, {
+          class: 'repos-container',
+        });
+        const contributorContainer = createAndAppend('section', mainContainer, {
+          class: 'contributor-container',
+        });
+        repos
+          .sort((a, b) => {
+            return a.name.localeCompare(b.name);
+          })
+          .forEach((elem, index) => {
```

Comment:
A better name for the parameter `elem` is `repo`. After all, the `.forEach()` method is iterating through the (sorted) `repos` array, and a single element of that array is a `repo`.

Once you have used the correct name, it is easy to see that you can use the `repo` variable to set the `text` property:

```js
.forEach((repo, index) => {
  createAndAppend('option', select, {
    value: index,
    text: repo.name,
    class: 'select-class',
  });
});
```

Reply:
elem is converted into repo, sir.